oryx-tui/src/cli.rs

```diff
@@ -0,0 +1,80 @@
+use clap::{
+    arg,
+    builder::ValueParser,
+    crate_description, crate_version,
+    error::{ContextValue, ErrorKind},
+    Command,
+};
+
+use crate::interface::NetworkInterface;
+
+fn parse_interface(interface: &str) -> Result<String, clap::Error> {
+    let interfaces = NetworkInterface::list()
+        .iter()
+        .map(|interface| interface.name.clone())
+        .collect::<Vec<String>>();
+
+    if interfaces.contains(&interface.to_string()) {
+        Ok(interface.to_string())
+    } else {
+        let mut err = clap::Error::new(ErrorKind::ValueValidation);
+        err.insert(
+            clap::error::ContextKind::ValidValue,
+            ContextValue::Strings(interfaces),
+        );
+        Err(err)
+    }
+}
+
+pub fn cli() -> Command {
```

Comment:
How about using a `derive` implementation here? See <https://docs.rs/clap/latest/clap/_derive/>

If you define a struct then you don't need to do custom string parsing yourself.

In other words, instead of going over the parsed values and doing `TransportProtocol::from_str(protocol).unwrap()`, you can let `clap` handle that for you at parse-time: 

```rust
pub struct Cli {
    #[arg(
        short = 't', 
        long, 
        help = "Transport layer protocols", 
        // ....
    )]
    pub transport: Vec<TransportProtocol>,
    /// other args
}
```

then simply do `Cli::parse()`.

Reply:
Can not do it, cause of the `no_std` :/